src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -34,91 +34,167 @@ def self.compile_ast(code, compile_options, boltlib_path)
           end
 
           load_bolt()
+          if compile_options.has_key?('versioned_project')
+            compile_for_project_plan(code, compile_options, boltlib_path, multithreaded)
+          else
+            compile_for_environment_plan(code, compile_options, boltlib_path)
+          end
+        else
+          compile_for_environment(code, compile_options, boltlib_path)
+        end
+      end
+      private_class_method :compile_ast
 
-          Puppet[:node_name_value] = compile_options['certname']
-
-          # Prior to PE-29443 variables were in a hash. Serialization between ruby/clojure/json did
-          # not preserver hash order necessary for deserialization. The data strucutre is now stored
-          # in a list for moving data and the list is used to construct an ordered ruby hash.
-          plan_variables = if compile_options['variables']['values'].is_a?(Array)
-                        compile_options['variables']['values'].each_with_object({}) do |param_hash, acc|
-                          acc[param_hash.keys.first] = param_hash.values.first
-                        end
-                      else
-                        compile_options['variables']['values']
-                      end
-
-          target_variables = compile_options.dig('target_variables', 'values') || {}
-
-          variables = {
-            variables: plan_variables,
-            target_variables: target_variables,
-          }
-
-          env_conf = {
-            pre_modulepath: boltlib_path,
-            envpath: Puppet[:environmentpath],
-            facts: compile_options['facts']['values'],
+      def self.compile_for_project_plan(code, compile_options, boltlib_path, multithreaded)
+
+        original_hiera_config = Puppet[:hiera_config]
+
+        Puppet[:node_name_value] = compile_options['certname']
```

Comment:
Do we not need to reset this back to original?

Reply:
A comment couldnt hurt around why we have to modify these settings directly instead of with Puppet.override :grimacing: